File: CS/DXExample.Module/Company.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace DXExample.Module {
    [DefaultClassOptions]
    public class Company : BaseObject {
        public Company(Session session) : base(session) { }
        private string _CompanyName;
        public string CompanyName {
            get { return _CompanyName; }
            set { SetPropertyValue("CompanyName", ref _CompanyName, value); }
        }
        private Address _CompanyAddress;
        public Address CompanyAddress {
            get { return _CompanyAddress; }
            set { SetPropertyValue("CompanyAddress", ref _CompanyAddress, value); }
        }
        private PhoneNumber _PhoneNumber;
        public PhoneNumber PhoneNumber {
            get { return _PhoneNumber; }
            set { SetPropertyValue("PhoneNumber", ref _PhoneNumber, value); }
        }
        [Association("Company-Users")]
        public XPCollection<CustomUser> Users {
            get { return GetCollection<CustomUser>("Users"); }
        }
        
    }

}

File: VB/DXExampleVB/DXExample.Module/Company.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace DXExample.Module
	<DefaultClassOptions> _
	Public Class Company
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _CompanyName As String
		Public Property CompanyName() As String
			Get
				Return _CompanyName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CompanyName", _CompanyName, value)
			End Set
		End Property
		Private _CompanyAddress As Address
		Public Property CompanyAddress() As Address
			Get
				Return _CompanyAddress
			End Get
			Set(ByVal value As Address)
				SetPropertyValue("CompanyAddress", _CompanyAddress, value)
			End Set
		End Property
		Private _PhoneNumber As PhoneNumber
		Public Property PhoneNumber() As PhoneNumber
			Get
				Return _PhoneNumber
			End Get
			Set(ByVal value As PhoneNumber)
				SetPropertyValue("PhoneNumber", _PhoneNumber, value)
			End Set
		End Property
		<Association("Company-Users")> _
		Public ReadOnly Property Users() As XPCollection(Of CustomUser)
			Get
				Return GetCollection(Of CustomUser)("Users")
			End Get
		End Property

	End Class

End Namespace
